components/public-api-server/pkg/apiv1/pagination.go

```diff
@@ -36,3 +36,22 @@ func paginationToDB(p *v1.Pagination) db.Pagination {
 		PageSize: int(validated.GetPageSize()),
 	}
 }
+
+func pageFromResults[T any](results []T, p *v1.Pagination) []T {
```

Comment:
We should probably add unit tests for this function in `pagination_test.go` just as we have for `validatePagination`.

Reply:
added